app/src/main/java/io/github/droidkaigi/confsched/util/AlarmUtil.java

```diff
@@ -0,0 +1,43 @@
+package io.github.droidkaigi.confsched.util;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+
+import org.parceler.Parcels;
+
+import java.util.Calendar;
+import java.util.Date;
+
+import io.github.droidkaigi.confsched.model.Session;
+import io.github.droidkaigi.confsched.receiver.SessionScheduleReceiver;
+
+public class AlarmUtil {
+    public static void registerSessionAlarm(Context context, Session session){
+        PendingIntent sender = buildSessionAlarmSender(context, session);
+
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+
+        long triggerAtMillis = calculateStartNotifyTime(session);
+        alarmManager.set(AlarmManager.RTC, triggerAtMillis, sender);
+    }
+
+    public static void unregisterSessionAlarm(Context context, Session session) {
+        PendingIntent sender = buildSessionAlarmSender(context, session);
+        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+        alarmManager.cancel(sender);
+    }
+
+    private static PendingIntent buildSessionAlarmSender(Context context, Session session) {
+        Intent intent = new Intent(context, SessionScheduleReceiver.class);
+        intent.putExtra(Session.class.getSimpleName(), Parcels.wrap(session));
+        return PendingIntent.getBroadcast(context, session.id, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+
+    }
+
+    private static long calculateStartNotifyTime(Session session) {
+        // 5minutes early
+        return session.stime.getTime() - 300000;
```

Comment:
I guess it is better to use `TimeUnit.MINUTES.toMillis(5)`, and static variable like this.

```java
private static final long REMIND_DURATION_MINUTES = TimeUnit.MINUTES.toMillis(5);

private static long calculateStartNotifyTime(Session session) {
    return session.stime.getTime() - REMIND_DURATION_MINUTES;
}
```